### DISCUSSION TITLE
Removal of Expo

### DISCUSSION THREAD

**Discussion by 7otion** [infinitered/ignite#2352] [Q&A] [answered]
I really don't want to use Expo but boilerplate comes with full support for it. Removing them from demo code, package.json, webpack conf and etc. is one thing but as far as i know expo also installs bunch of heavy native packages that will never be used for vanilla react projects. is there a way for me to avoid this ? Boilerplate was used to be expo independent why did they chose this path ?

**Comment by comertcimen** [comment]
The project includes App.js and index.js at the root level. Depending on using expo or react native cli, the project starts that way. You can still use expo modules in bare app and using expo sdk depends on your needs.

**Reply by adirzoari** [reply]
@ev3nst did you find a way to do that? I don't want use Expo too. just react-native cli

**Comment by frankcalise** [comment, marked as answer]
@ev3nst You might be thinking of older Expo, however as per their docs 
> The expo package has a small footprint; it includes only a minimal set of packages that are needed in nearly every app and the module and autolinking infrastructure that other Expo SDK packages are built with.

Also, the boilerplate was changed because it represents the way we start off client projects. Although we expose it out to the community, we build Ignite around how Infinite Red does work. So it may not fit your needs exactly, but give you a great starting point and then you can customize as needed.

**Reply by 7otion** [reply]
*we build Ignite around how Infinite Red does work*

I understand if thats how Red wants to go forward. I personally do not like Expo at all. I will continue to use other boilerplates available for now. But also i feel like i have to say that If **Red** would provide 2 sets of boilerplate with/without expo i would immedietaly switch to this repo again but thats almost twice the workload so again i understand your angle and wish you guys the best.

**Reply by jamonholmgren** [reply]
@ev3nst Can I have you explain more about your dislike of Expo? Is it primarily the Expo services, like EAS, Expo Go, etc? Or do you not want to use any Expo open source packages at all? Very curious about this perspective. (And any other info as to why you don't like Expo is helpful!)

**Comment by frankcalise** [comment]
@ev3nst Understood, we're still exploring on how to best incorporate these types of "template" options where it's useful for your scenario but also easy enough for us to maintain.

However, you might find this interesting for your use case: https://github.com/frankcalise/ignite-vanilla-template

I started a truly Expo-less template, but there are no guarantees here as it's still very much a work in progress. I had to rip out localization for the time being and still working on getting react native web to work properly. But if it helps, feel free to start there.

**Reply by adirzoari** [reply]
@frankcalise it there any cli to install your project? it so useful without the Expo if I don't use it.

**Reply by frankcalise** [reply]
There's no cli for it necessarily but you can clone the repo and get started from there.

**Reply by adirzoari** [reply]
Can I create a new app with a new name using this? or should I change it manually?

**Reply by frankcalise** [reply]
Probably use something like https://www.npmjs.com/package/react-native-rename
